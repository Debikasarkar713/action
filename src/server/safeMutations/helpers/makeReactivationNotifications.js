import {REJOIN_TEAM} from 'universal/utils/constants';


const makeReactivationNotifications = (notifications, reactivatedUsers, teamMembers, inviter) => {
  const {teamName, userId} = inviter;
  const restOfTeamUserIds = teamMembers
    .filter((m) => m.isNotRemoved === true && m.id !== userId)
    .map((m) => m.userId);
  const notificationsToSend = {};
  reactivatedUsers.forEach((user, idx) => {
    const notification = notifications[idx];
    const {preferredName, id: reactivatedUserId} = user;

    // make a notification to the person being reactivated
    notificationsToSend[reactivatedUserId] = [notification];

    // make a notification for the other team members annoucing the reactivation
    const rejoinNotification = {
      teamName,
      preferredName,
      type: REJOIN_TEAM
    };
    restOfTeamUserIds.forEach((notificationUserId) => {
      if (inviter.userId === notificationUserId) return;
      notificationsToSend[notificationUserId] = notificationsToSend[notificationUserId] || [];
      notificationsToSend[notificationUserId].push(rejoinNotification);
    });
  });
  return notificationsToSend;
};

export default makeReactivationNotifications;
