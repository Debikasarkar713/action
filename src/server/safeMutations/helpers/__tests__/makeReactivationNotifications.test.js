import makeReactivationNotifications from 'server/safeMutations/helpers/makeReactivationNotifications';
import {__now} from 'server/__tests__/setup/mockTimes';
import MockDate from 'mockdate';

MockDate.set(__now);
console.error = jest.fn();

describe('makeReactivationNotifications', () => {
  test('creates a notification for the reactivated user giving them a new authToken', () => {
    // SETUP
    const notifications = [{
      id: 1
    }];
    const reactivatedUsers = [{
      id: 1,
      tms: ['team123']
    }];
    const inviter = {teamId: 'team456', teamName: 'The 456s', inviterName: 'Four', userId: 4};

    // TEST
    const result = makeReactivationNotifications(notifications, reactivatedUsers, [], inviter);
    // VERIFY
    expect(result).toMatchSnapshot();
  });
  test('creates a notification for the team (less the inviter)', () => {
    // SETUP
    const reactivatedUsers = [{
      id: 1,
      tms: ['team123'],
      preferredName: 'One'
    }];
    const notifications = [{
      id: 1
    }];
    const teamMembers = [
      {id: '2:team456', preferredName: 'Two', userId: 2, isNotRemoved: true},
      {id: '3:team456', preferredName: 'Three', userId: 3, isNotRemoved: true},
      {id: '4:team456', preferredName: 'Four', userId: 4, isNotRemoved: true},
      {id: '5:team456', preferredName: 'Five', userId: 5, isNotRemoved: false}
    ];
    const inviter = {teamId: 'team456', teamName: 'The 456s', inviterName: 'Four', userId: 4};

    // TEST
    const result = makeReactivationNotifications(notifications, reactivatedUsers, teamMembers, inviter);
    // VERIFY
    expect(result).toMatchSnapshot();
  });
});

