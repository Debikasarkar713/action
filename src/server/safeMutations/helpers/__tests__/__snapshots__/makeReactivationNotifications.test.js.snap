// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeReactivationNotifications creates a notification for the reactivated user giving them a new authToken 1`] = `
Object {
  "1": Array [
    Object {
      "id": 1,
    },
  ],
}
`;

exports[`makeReactivationNotifications creates a notification for the team (less the inviter) 1`] = `
Object {
  "1": Array [
    Object {
      "id": 1,
    },
  ],
  "2": Array [
    Object {
      "preferredName": "One",
      "teamName": "The 456s",
      "type": "REJOIN_TEAM",
    },
  ],
  "3": Array [
    Object {
      "preferredName": "One",
      "teamName": "The 456s",
      "type": "REJOIN_TEAM",
    },
  ],
}
`;
